File: Mobile/app/src/main/java/com/example/iccms_mobile/data/models/Quotation.kt
package com.example.iccms_mobile.data.models

import com.google.gson.annotations.SerializedName

data class Quotation(
    @SerializedName("quotationId")
    val quotationId: String = "",

    @SerializedName("projectId")
    val projectId: String = "",

    @SerializedName("maintenanceRequestId")
    val maintenanceRequestId: String = "",

    @SerializedName("clientId")
    val clientId: String = "",

    @SerializedName("contractorId")
    val contractorId: String = "",

    @SerializedName("adminApproverUserId")
    val adminApproverUserId: String = "",

    @SerializedName("description")
    val description: String = "",

    @SerializedName("total")
    val total: Double = 0.0,

    @SerializedName("totalWithMarkup")
    val totalWithMarkup: Double = 0.0,

    @SerializedName("status")
    val status: String = "",

    @SerializedName("validUntil")
    val validUntil: String = "",

    @SerializedName("createdAt")
    val createdAt: String = "",

    @SerializedName("sentAt")
    val sentAt: String? = null,

    @SerializedName("approvedAt")
    val approvedAt: String? = null,

    @SerializedName("items")
    val items: List<QuotationItem> = emptyList(),

    @SerializedName("subtotal")
    val subtotal: Double = 0.0,

    @SerializedName("taxTotal")
    val taxTotal: Double = 0.0,

    @SerializedName("taxTotalWithMarkup")
    val taxTotalWithMarkup: Double = 0.0,

    @SerializedName("grandTotal")
    val grandTotal: Double = 0.0,

    @SerializedName("markupRate")
    val markupRate: Double = 1.0,

    @SerializedName("currency")
    val currency: String = "ZAR",

    @SerializedName("adminApprovedAt")
    val adminApprovedAt: String? = null,

    @SerializedName("clientRespondedAt")
    val clientRespondedAt: String? = null,

    @SerializedName("clientDecisionNote")
    val clientDecisionNote: String? = null,

    @SerializedName("updatedAt")
    val updatedAt: String = "",

    @SerializedName("isAiGenerated")
    val isAiGenerated: Boolean = false,

    @SerializedName("estimateId")
    val estimateId: String? = null,

    @SerializedName("pmEditedAt")
    val pmEditedAt: String? = null,

    @SerializedName("pmEditNotes")
    val pmEditNotes: String? = null,

    @SerializedName("pmRejectedAt")
    val pmRejectedAt: String? = null,

    @SerializedName("pmRejectReason")
    val pmRejectReason: String? = null
)

File: API/ICCMS-API/Models/Quotation.cs
using Google.Cloud.Firestore;

namespace ICCMS_API.Models
{
    [FirestoreData]
    public class Quotation
    {
        [FirestoreProperty("quotationId")]
        public string QuotationId { get; set; } = string.Empty;

        [FirestoreProperty("projectId")]
        public string ProjectId { get; set; } = string.Empty;

        [FirestoreProperty("maintenanceRequestId")]
        public string MaintenanceRequestId { get; set; } = string.Empty;

        [FirestoreProperty("clientId")]
        public string ClientId { get; set; } = string.Empty;

        [FirestoreProperty("contractorId")]
        public string ContractorId { get; set; } = string.Empty;

        [FirestoreProperty("adminApproverUserId")]
        public string AdminApproverUserId { get; set; } = string.Empty;

        [FirestoreProperty("description")]
        public string Description { get; set; } = string.Empty;

        [FirestoreProperty("total")]
        public double Total { get; set; }

        [FirestoreProperty("status")]
        public string Status { get; set; } = string.Empty;

        [FirestoreProperty("validUntil")]
        public DateTime ValidUntil { get; set; }

        [FirestoreProperty("createdAt")]
        public DateTime CreatedAt { get; set; }

        [FirestoreProperty("sentAt")]
        public DateTime? SentAt { get; set; }

        [FirestoreProperty("approvedAt")]
        public DateTime? ApprovedAt { get; set; }

        // New fields for enhanced workflow
        [FirestoreProperty("items")]
        public List<QuotationItem> Items { get; set; } = new List<QuotationItem>();

        [FirestoreProperty("subtotal")]
        public double Subtotal { get; set; }

        [FirestoreProperty("taxTotal")]
        public double TaxTotal { get; set; }

        [FirestoreProperty("grandTotal")]
        public double GrandTotal { get; set; }

        [FirestoreProperty("markupRate")]
        public double MarkupRate { get; set; } = 1.0; // Default to no markup (1.0 = 100%)

        [FirestoreProperty("currency")]
        public string Currency { get; set; } = "ZAR";

        [FirestoreProperty("adminApprovedAt")]
        public DateTime? AdminApprovedAt { get; set; }

        [FirestoreProperty("clientRespondedAt")]
        public DateTime? ClientRespondedAt { get; set; }

        [FirestoreProperty("clientDecisionNote")]
        public string? ClientDecisionNote { get; set; }

        [FirestoreProperty("updatedAt")]
        public DateTime UpdatedAt { get; set; }

        // New fields for AI workflow
        [FirestoreProperty("isAiGenerated")]
        public bool IsAiGenerated { get; set; } = false;

        [FirestoreProperty("estimateId")]
        public string? EstimateId { get; set; }

        [FirestoreProperty("pmEditedAt")]
        public DateTime? PmEditedAt { get; set; }

        [FirestoreProperty("pmEditNotes")]
        public string? PmEditNotes { get; set; }

        [FirestoreProperty("pmRejectedAt")]
        public DateTime? PmRejectedAt { get; set; }

        [FirestoreProperty("pmRejectReason")]
        public string? PmRejectReason { get; set; }
    }
}
